cmd/run.go

```diff
@@ -172,6 +172,25 @@ func runRun(cmd *cobra.Command, newClient ClientFactory) (err error) {
 	client, done := newClient(ClientConfig{Verbose: cfg.Verbose}, clientOptions...)
 	defer done()
 
+	var (
+		digested        bool
+		validUndigested bool
+		justBuilt       bool
+	)
+
+	// check for digested image first
+	digested, err = isDigested(cfg.Image)
+	if err != nil {
+		return err
+	}
+
+	if !digested {
+		validUndigested, err = isUndigested(cfg.Image)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Hmm.. returning an error here would mean some complications after combining with `isDigested()`. This function is called during run and validation (`func (c deployConfig) Validate(cmd *cobra.Command) (err error)`) and our deploy printer (`func printDeployMessages(out io.Writer, f fn.Function)`). Now that I think of it the printer should probably not check the return value of `err` but, the validator should. 
Here comes the scenario:
`func create -l go` -> `func deploy --registry=docker.io/yourregistry` -> this gives you an error of empty string simply because no image name was captured for the deploy config because the image was not yet built and its name wasnt even created yet.